### PR TITLE
Add keyboard control robot code for testing and demos

### PR DESCRIPTION
**Pull request by PeterJCLaw**
This aims to provide a way to explore robot behaviour without needing to write code, for various purposes.

The changes here incorporate the work done in the private repo (specifically https://github.com/srobo/competition-simulator-private/pull/6 and https://github.com/srobo/competition-simulator-private/pull/8), rebased onto the HEAD of this repo to avoid conflicts and remove redundant/unnecessary changes.

Testing for this was manual, it's not obvious how we could test this in an automatic fashion.

### CODE REVIEW

File: controllers/example_controller/keyboard_controller.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
+from sr.robot import Robot
+from controller import Keyboard
+
+TIMESTEP = 16
+NO_KEY_PRESSED = -1
+
+
+def print_distance_sensors(robot: Robot) -> None:
+    distance_sensor_names = [
```
Comment (RealOrangeOne) [inline controllers/example_controller/keyboard_controller.py:9]: Should we just import this instead?

File: controllers/example_controller/keyboard_controller.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
+from sr.robot import Robot
+from controller import Keyboard
+
+TIMESTEP = 16
```
Comment (RealOrangeOne) [inline controllers/example_controller/keyboard_controller.py:4]: We must be able to derive this from somewhere?

File: controllers/example_controller/keyboard_controller.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+            elif key == ord('A'):
+                R.motors[0].m0.power = -25
+                R.motors[0].m1.power = 25
+
+            elif key == ord('D'):
+                R.motors[0].m0.power = 25
+                R.motors[0].m1.power = -25
+
+            elif key == ord('E'):
```
Comment (RealOrangeOne) [inline controllers/example_controller/keyboard_controller.py:62]: "E" feels annoying to press here - should it just be space?

File: controllers/example_controller/keyboard_controller.py (hunk 1 above)
Comment (PeterJCLaw) [reply controllers/example_controller/keyboard_controller.py:9]: I considered that, however they're not the same names.

File: controllers/example_controller/keyboard_controller.py (hunk 2 above)
Comment (PeterJCLaw) [reply controllers/example_controller/keyboard_controller.py:4]: Maybe? Bear in mind that this seems to be expected to be different to the basic timestemp. Perhaps @JennyLynnFletcher can provide more context?

Comment (WillB97) [reply controllers/example_controller/keyboard_controller.py:4]: This is simply how often the control input is sampled.

Comment (PeterJCLaw) [reply controllers/example_controller/keyboard_controller.py]: d0e9162

**Review by RealOrangeOne — APPROVED** [review]
This PR as a basis I think looks fine. It'll definitely be useful to use. I'd like to add an API into the `Robot` class itself to support some control like this, but I that's something for a further PR.

**Pull request merged**